src/EditorFeatures/CSharpTest/Classification/TotalClassifierTests.cs

```diff
@@ -3141,5 +3144,114 @@ void M()
+            using var workspace = EditorTestWorkspace.CreateCSharp(""""
+                using System.Text.RegularExpressions;
+
+                class C
+                {
+                    // class D { }
+                    void M()
+                    {
+                        new Regex("(a)");
+                        var s1 = "s1";
+                        var s2 = $"s2";
+                        var s3 = @"s3";
+                        var s4 = """
+                        s4
+                        """;
+                    }
+                }
+                """");
+            var document = workspace.Documents.First();
+
+            var listenerProvider = workspace.ExportProvider.GetExportedValue<IAsynchronousOperationListenerProvider>();
+            var globalOptions = workspace.ExportProvider.GetExportedValue<IGlobalOptionService>();
+
+            var provider = new CopyPasteAndPrintingClassificationBufferTaggerProvider(
+                workspace.GetService<IThreadingContext>(),
+                workspace.GetService<ClassificationTypeMap>(),
+                listenerProvider,
+                globalOptions);
+
+            var buffer = document.GetTextBuffer();
+            using var tagger = provider.CreateTagger<IClassificationTag>(buffer);
+
+            var allCode = buffer.CurrentSnapshot.GetText();
+            var tags = tagger!.GetAllTags(new NormalizedSnapshotSpanCollection(buffer.CurrentSnapshot.GetFullSpan()), CancellationToken.None);
+
+            var actualOrdered = tags.OrderBy((t1, t2) => t1.Span.Span.Start - t2.Span.Span.Start);
+
+            var actualFormatted = actualOrdered.Select(a => new FormattedClassification(allCode.Substring(a.Span.Span.Start, a.Span.Span.Length), a.Tag.ClassificationType.Classification));
+
+            AssertEx.Equal(new[]
+            {
+                Keyword("using"),
+                Namespace("System"),
+                Identifier("System"),
+                Operators.Dot,
+                Namespace("Text"),
+                Identifier("Text"),
+                Operators.Dot,
+                Namespace("RegularExpressions"),
+                Identifier("RegularExpressions"),
+                Punctuation.Semicolon,
+                Keyword("class"),
+                Class("C"),
+                Punctuation.OpenCurly,
+                Comment("// class D { }"),
+                Keyword("void"),
+                Method("M"),
+                Punctuation.OpenParen,
+                Punctuation.CloseParen,
+                Punctuation.OpenCurly,
+                Keyword("new"),
+                Class("Regex"),
+                Identifier("Regex"),
+                Punctuation.OpenParen,
+                String("\""),
+                Regex.Grouping("("),
+                Regex.Text("a"),
+                Regex.Grouping(")"),
+                String("\""),
+                Punctuation.CloseParen,
+                Punctuation.Semicolon,
+                Keyword("var"),
```

Comment:
>Keyword("var"), [](http://example.com/codeflow?start=16&length=15)

out of curiosity, why is Keyword("var") duplicated?

Reply:
so currently we do not dedupe between tags created by both the underlying syntactic tagger (innacurate, but fast) and the semantic tagger (accurate, but slow).  I will do a followup to do that on top of this PR.